core/amber/src/main/python/core/runnables/network_sender.py

```diff
@@ -85,3 +88,14 @@ def _send_control(
         """
         python_control_message = PythonControlMessage(tag=to, payload=control_payload)
         self._proxy_client.call_action("control", bytes(python_control_message))
+
+    async def heartbeat(self, interval, stop_event):
+        while not stop_event.is_set():
+            try:
+                # Send a heartbeat to the server
+                self._proxy_client.do_action(flight.Action("heartbeat", b""))
```

Comment:
for encapsulation purpose, network sender better not see the underlying "Flight". Can we use `call_action` instead? We can discuss it.